File: frontend/src/app/register/viewModels/vehicle.view-model.ts


export class VehicleViewModel {
    id: number;
    brand: string;
    model: string;
    color: string;
    year: number;
    price: number;
    description: string;
    is_new: boolean;


    constructor(v: any) {
        this.id = v.id != null ? v.id : null;
        this.brand = v.brand != null ? v.brand : null;
        this.color = v.color != null ? v.color : null;
        this.model = v.model != null ? v.model : null;
        this.year = v.year ? v.year : null;
        this.price = v.price ? v.price : null;
        this.description = v.description ? v.description : null;
        this.is_new = v.is_new != null ? v.is_new : null;
    }
}

File: backend/Teste.App/viewModel/VehicleViewModel.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Text;

namespace Teste.App.viewModel
{
    [DataContract]
    public class VehicleViewModel
    {

        [DataMember(Name = "id")]
        public int? Id { get; set; }

        [DataMember(Name = "brand")]
        public string Brand { get; set; }

        [DataMember(Name = "model")]
        public string Model { get; set; }

        [DataMember(Name = "color")]
        public string Color { get; set; }

        [DataMember(Name = "year")]
        public int Year { get; set; }

        [DataMember(Name = "price")]
        public Decimal Price { get; set; }

        [DataMember(Name = "description")]
        public string Description { get; set; }

        [DataMember(Name = "is_new")]
        public Boolean IsNew { get; set; }

        [IgnoreDataMember]
        public DateTime? AddedDate { get; set; }

        [DataMember(Name = "modified_date")]
        public DateTime? ModifiedDate { get; set; }
    }
}
